shoot-client/network-connection.sh

```diff
@@ -44,98 +46,124 @@ function configure_tcp() {
   set_value /proc/sys/net/ipv4/tcp_retries2 $tcp_retries2
 }
 
-if [[ -z "$DO_NOT_CONFIGURE_KERNEL_SETTINGS" ]]; then
+function configure_bonding() {
+  local addr
+  local targets
+
+  if [[ "$IS_SHOOT_CLIENT" == "true" ]]; then
+    # IP address is fixed on shoot side
+    addr="$bondPrefix.$((vpn_client_index+10))/24"
+    targets="${bondPrefix}.1" # using a dummy address as kube-apiserver IPs are unknown
+  else
+    # for each kube-apiserver pod acquire an IP via consensus
+    # based on pod annotations (details see go part)
+    log "acquiring ip address for bonding"
+    OUTPUT=/tmp/acquired-ip ./acquire-ip
+    addr="$(</tmp/acquired-ip)/24"
+
+    for ((i=0; i < $HA_VPN_CLIENTS; i++))
+    do
+      if (( i > 0 )); then
+        targets+=','
+      fi
+      targets+="${bondPrefix}.$((i+10))"
+    done
+  fi
+  log "bonding address is $addr"
+
+  ip link del bond0 2> /dev/null || true
+  # use bonding with active-backup mode and activate ARP link monitoring
+  local cmd=$(echo ip link add bond0 type bond mode active-backup fail_over_mac 1 arp_interval 1000 arp_ip_target \"$targets\" arp_all_targets 0)
+  log $cmd
+  $(eval echo $cmd)
+  for ((i=0; i < $HA_VPN_SERVERS; i++))
```

Comment:
Probably analogous to `HA_VPN_CLIENTS`.

Reply:
Only used in the function `configure_bonding`, see above